karabo_bridge/simulation.py

```diff
@@ -322,21 +334,32 @@ def run(self):
         poller = zmq.Poller()
         poller.register(self.server_socket, zmq.POLLIN)
```

Comment:
I think you need to register it with `POLLOUT` as well, otherwise you'll never get an event for PUB or PUSH sockets.

The code below will still work because it doesn't check for those events, but it means it will always have 100 ms sleep before trying to generate the next data. If we want rate limiting, ideally we should do it in a way that accounts for the time taken to generate and send a train's worth of data.

Reply:
That's a good point, I'll try to improve that (I to get closer to a real system where you have delays between trains and also trains can be skipped if the client does not request data, i.e. get trainId gaps).